tests/PHPStan/Analyser/ArrayIsListTest.php

```diff
@@ -0,0 +1,25 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Analyser;
+
+use PhpParser\Node;
+use PHPStan\Testing\TypeInferenceTestCase;
+
+class ArrayIsListTest extends TypeInferenceTestCase
+{
+
+	public function testNotList(): void
+	{
+		$this->processFile(
+			__DIR__ . '/data/array-is-list.php',
+			function (Node $node, Scope $scope): void {
+				if (!$node instanceof Node\Stmt\Return_ || $node->expr === null) {
+					return;
+				}
+				$type = $scope->getType($node->expr);
+				$this->assertFalse($type->isList()->yes());
```

Comment:
This test is weird, no test in phpstan-src looks like this :) Please write a test that tests type acceptance, so for example CallMethodsRuleTest, or ReturnTypeRuleTest. Thank you!

Reply:
Apologies, I struggled with writing a classical test for this, but I can't remember why. I now converted it to a test in ReturnTypeRuleTest.